File: connection-backend/src/main/java/de/florianschmitt/model/rest/ERequestDTO.java
package de.florianschmitt.model.rest;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Set;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import org.hibernate.validator.constraints.Email;
import org.hibernate.validator.constraints.NotBlank;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class ERequestDTO implements Serializable {

    private String requestIdentifier;

    @NotNull
    @Size(min = 1)
    private Set<Long> languageIds;

    @NotNull
    private LocalDateTime datetime;

    @NotBlank
    private String ocation;

    @NotBlank
    private String street;

    @NotBlank
    private String postalCode;

    @NotBlank
    private String city;

    @NotBlank
    @Email
    private String email;

    @NotBlank
    private String phone;

    private String state;

    private LocalDateTime createdAt;

    private String acceptedByVolunteer;
}

File: src/main/kotlin/de/florianschmitt/model/rest/ERequestDTO.kt
package de.florianschmitt.model.rest

import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank

import javax.validation.constraints.NotNull
import javax.validation.constraints.Size
import java.io.Serializable
import java.time.LocalDateTime
import javax.persistence.Column

class ERequestDTO : Serializable {

    var requestIdentifier: String? = null

    @NotNull
    @Size(min = 1)
    var languageIds: Set<Long>? = null

    var datetime: LocalDateTime? = null

    var dateDescription: String? = null

    @NotBlank
    var occasionEnum: String? = null

    var occasionString: String? = null

    @NotBlank
    var street: String? = null

    @NotBlank
    var postalCode: String? = null

    @NotBlank
    var city: String? = null

    @NotBlank
    var requesterName: String? = null

    var requesterInstitution: String? = null

    @NotBlank
    @Email
    var email: String? = null

    @NotBlank
    var phone: String? = null

    var state: String? = null

    var createdAt: LocalDateTime? = null

    var acceptedByVolunteer: String? = null
}
